src/Disks/S3/DiskS3.cpp

```diff
@@ -902,12 +922,371 @@ void DiskS3::shutdown()
     client->DisableRequestProcessing();
 }
 
-std::optional<DiskS3::ObjectMetadata> DiskS3::createObjectMetadata(const String & path) const
+void DiskS3::createFileOperationObject(const String & operation_name, UInt64 revision, const DiskS3::ObjectMetadata & metadata)
 {
-    if (send_metadata)
-        return (DiskS3::ObjectMetadata){{"path", path}};
+    const String key = "operations/r" + revisionToString(revision) + "-" + operation_name;
```

Comment:
Should it support `remove*` operations? Now It can be only move or hardlink. Why we can omit other operations?

Reply:
Yes, it could seem that we need to support `remove` operations as well, but we can actually skip it. 
Before every removing operation CH renames the part directory to '_delete_tmp_XXX'.
So we need only to restore this rename to '_delete_tmp_XXX'.
When CH starts it will just remove all these '_delete_tmp_XXX' directories by himself